### ISSUE TITLE
Nginx HSTS configuration error

### ISSUE THREAD

**Issue by Gregory92** [nextcloud/documentation#6744]
Hello,

After installing nextcloud 21.0.2 on a fresh Ubuntu 20.04 LTS, I could not get the The [“Strict-Transport-Security” HTTP header is not configured"](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/security_setup_warnings.html#the-strict-transport-security-http-header-is-not-configured) error message to go away on the admin warning page. I verified that the option in the config file I got from [docs](https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html#nextcloud-in-a-subdir-of-the-nginx-webroot) was uncommented, tried removing and adding preload again, etc. to no avail.

While researching nginx [add_headers](https://www.nginx.com/blog/http-strict-transport-security-hsts-and-nginx/) directives, I discovered that the HSTS directive, while added at the top of the server block, will **not** work/apply to the */nextcloud* location block because the */nextcloud* block has *add_headers* directives inside it as well.

Basically, if you declare the HSTS option at the top of the server block in Nginx, it will be inherited into all location blocks beneath it, **unless** one of those location blocks also contains *add_header* directives, in which case it will need to be redeclared. Once I redeclared the header inside the */nextcloud* block in addition to the one already at the top and reloaded nginx, boom, error message gone.

I tried to change the docs myself, but GitHub asked me to fork the repository to make changes which I did not feel comfortable doing or really know how, so I opened this bug report.

Let me know if you guys need any more information.

Thank you,

Gregory 

Ps. Keep up the great work, 21 is a huge improvement coming from 20 on my machine!

**Comment by HouraisanNEET** [comment]
Hi there, I submitted a possible fix, which may interest you.

**Comment by HouraisanNEET** [comment]
Hi there, there is already a patch submitted. However it takes some time to be approved.
#6408
